Add traceId and spanId properties in LogstashJsonFormatter output

tbdrake opened the pull request:
Fixes #53 

See related changes in https://github.com/serilog/serilog/pull/1955

For reference, `Serilog.Formatting.Json.JsonFormatter` also includes these properties here:
https://github.com/serilog/serilog/blob/cd6420aeafcdedd3790da1a03a419e389f0faa64/src/Serilog/Formatting/Json/JsonFormatter.cs#L74-L84

~~Also enabled `<Nullable>` in the unit test project (already enabled in the main project) and fixed some warnings.~~ moved to #56 and #57.

nojaf commented:
@smaug123, I don't see any harm in this, do you?

tbdrake commented:
@nojaf One concern I realized is that existing code could be adding `traceId` and `spanId`, for example with https://github.com/RehanSaeed/Serilog.Enrichers.Span, and in that case the `LogstashJsonFormatter` would have duplicate properties. I found this issue https://github.com/serilog-contrib/Serilog.Sinks.Network/issues/39 so it looks like this applies to other properties like `message` as well.

Smaug123 commented:
I don't see any harm in this per se, but I'm certainly not a domain expert, and that issue does look concerning! (On general principles I'd prefer to pull out the `Nullable` enablement into a separate PR, so that the change with possibly-dangerous semantics can go in on its own.)

tbdrake commented:
> I don't see any harm in this per se, but I'm certainly not a domain expert, and that issue does look concerning! (On general principles I'd prefer to pull out the `Nullable` enablement into a separate PR, so that the change with possibly-dangerous semantics can go in on its own.)

Thank you

tbdrake commented:
> I added a failing test in the latest commit and plan to look into #39 and maybe open a separate PR.

I adjust the test to pass based on looking into it further and added [comment here](https://github.com/serilog-contrib/Serilog.Sinks.Network/issues/39#issuecomment-2800212524) on #39.

tbdrake commented:
@nojaf No rush but do you have any ETA for a next release, and do you think this change could go in it?

nojaf reviewed (approved):
Thanks, this looks fine. I'll publish a new alpha so you can test this further on your end.
